src/config/frontend/EdFi.DmsConfigurationService.Frontend.AspNetCore/Modules/ApplicationModule.cs

```diff
@@ -197,4 +214,77 @@ ILogger<ApplicationModule> logger
             _ => Results.Problem(statusCode: 500),
         };
     }
+
+    private static async Task<IResult> ResetCredential(
+        long id,
+        HttpContext httpContext,
+        IApplicationRepository repository,
+        IClientRepository clientRepository,
+        ILogger<ApplicationModule> logger
+    )
+    {
+        List<ApplicationCredentialsResponse> response = [];
+        var apiClientsResult = await repository.GetApplicationApiClients(id);
+        switch (apiClientsResult)
+        {
+            case ApplicationApiClientsResult.Success success:
+                foreach (var client in success.Clients)
+                {
+                    try
+                    {
+                        logger.LogInformation("Resetting client {clientId}", client.ClientId);
+                        var clientResetResult = await clientRepository.ResetCredentialsAsync(
+                            client.ClientUuid.ToString()
+                        );
+                        switch (clientResetResult)
+                        {
+                            case ClientResetResult.Success resetSuccess:
+                                response.Add(
+                                    new ApplicationCredentialsResponse()
+                                    {
+                                        Id = id,
+                                        Key = client.ClientId,
+                                        Secret = resetSuccess.ClientSecret,
+                                    }
+                                );
+                                break;
+                            case ClientResetResult.FailureUnknown failure:
+                                logger.LogError(
+                                    "Error resetting client credentials {clientId} {clientUuid}: {message}",
+                                    client.ClientId,
+                                    client.ClientUuid,
+                                    failure.FailureMessage
+                                );
+                                return Results.Problem(statusCode: 500);
+                        }
+                    }
+                    catch (Exception ex)
+                    {
+                        logger.LogError(
+                            "Error deleting client {clientId} {clientUuid}: {message}",
+                            client.ClientId,
+                            client.ClientUuid,
+                            ex.Message
+                        );
+                        return Results.Problem(statusCode: 500);
+                    }
+                }
+
+                break;
+            case ApplicationApiClientsResult.FailureUnknown failure:
+                logger.LogError("Error fetching ApiClients: {failure}", failure);
+                return Results.Problem(statusCode: 500);
+        }
+
+        switch (response.Count)
```

Comment:
Our data model supports many `ApiClient` per `Application` but the AdminApi spec doesn't.

Reply:
I'm going to talk to Jason and Vinaya about this. I don't understand the use case of 1-to-many on those database tables.

Reply:
For now I'm assuming that we do not need to distinguish `Application` and `ApiClient` from each other, like the `EdFi_Admin` database does.

Reply:
After discussion, leaving the many-to-one in the database but only using `First()` in `reset-credential`